packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -432,19 +432,30 @@ export class FluidDataStoreRuntime
 		return context.getChannel();
 	}
 
-	public createChannel(id: string = uuid(), type: string): IChannel {
+	public createChannel(channel: IChannel): void;
+	public createChannel(id: string | undefined, type: string): IChannel;
```

Comment:
Can you please add some documentation here (particularly clarifying the two possible calling patterns)?

Longer-term, do we think this alternate calling pattern makes its way into `IFluidDataStoreRuntime`?

Did you consider a separately-named method rather than an overload?  Overloads tend to be tougher to work with (consider the repeated conditionals below deducing how the method was called), so unless there's some key benefit to be had I'd probably prefer a unique name.  Especially considering the return type also differs.

Reply:
Another drawback to overloads is that they're a minor maintenance headache for consumers if/when APIs change. Uniquely named functions are easier to find/replace update safely in large codebases. They're also easier to auto-upgrade with tools like jscodeshift (e.g. imagine we shipped an upgrade tool that ran on your codebase and automatically updated your code).

@ChumpChief We should start a design pattern/code style doc somewhere. The info on overloads seems generally useful. Our stance on enums could also be written down there.

Reply:
+1; especially overloads with different return types feel like asking for trouble.

Reply:
Shifted to new api addchannel(channel): void.

We can add this api to the interface later in next branch.